service/misc_services.cc

```diff
@@ -214,12 +214,13 @@ future<lowres_clock::duration> cache_hitrate_calculator::recalculate_hitrates()
         });
     }).then([this] {
         _slen = _gstate.size();
+        using namespace std::chrono_literals;
         return _gossiper.add_local_application_state(gms::application_state::CACHE_HITRATES, gms::versioned_value::cache_hitrates(_gstate)).then([this] {
             // if max difference during this round is big schedule next recalculate earlier
             if (_diff < 0.01) {
-                return std::chrono::milliseconds(2000);
+                return lowres_clock::duration(2000ms);
```

Comment:
The surprising thing for me, is that if this `lowres_clock::duration(2000ms)` works (I assume you tested it), it means there **is** a convertor from `std::chrono::milliseconds` to `lowres_clock::duration`, so why wasn't it used implicitly? I guess C++ marks this convertor "explicit" which explains this.

Reply:
It's implicit if it loses resolution, and explicit if it doesn't lose resolution.